Source/EventFlow.Autofac.Tests/UnitTests/Aggregates/AggregateFactoryTests.cs

```diff
@@ -0,0 +1,103 @@
+﻿// The MIT License (MIT)
+//
+// Copyright (c) 2015 Rasmus Mikkelsen, Jaco Coetzee
+// https://github.com/rasmus/EventFlow
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of
+// this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to
+// use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+// the Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using Autofac;
+using EventFlow.Aggregates;
+using EventFlow.Autofac.Extensions;
+using EventFlow.Configuration;
+using EventFlow.TestHelpers.Aggregates.Test;
+using FluentAssertions;
+using NUnit.Framework;
+using EventFlow.Extensions;
+
+namespace EventFlow.Autofac.Tests.UnitTests.Aggregates
+{
+    [TestFixture]
+    public class AggregateFactoryTests
+    {
+        [Test]
+        public async void AutofacAggregateFactoryResolvesConstructorParameters()
+        {
+            // Arrange
+            var containerBuilder = new ContainerBuilder();
+            using (var resolver = EventFlowOptions.New
+                .UseAutofacContainerBuilder(containerBuilder)
+                .UseAutofacAggregateFactory()
+                .AddAggregateRoots(typeof(AggregateFactoryTests).Assembly)
+                .RegisterServices(f => f.RegisterType(typeof(Pinger)))
+                .CreateResolver())
+            {
+                // Arrange
+                var id = TestId.New;
```

Comment:
@JC008 Rename variables to something that reflects the type, e.g. `testAggregateWithResolver` for a type `TestAggregateWithResolver`. Its long, be improves readability slightly.